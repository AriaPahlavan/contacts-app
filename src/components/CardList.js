import React from 'react';
import Card from './Card';

const CardList = ({contacts}) => {
    return (
        <div>
            {contacts.map(x => <Card key={x.id}
                                     id={x.id}
                                     name={x.name}
                                     email={x.email}/>)}
        </div>
    )
}

export default CardList;
