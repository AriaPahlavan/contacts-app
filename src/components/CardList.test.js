import {shallow} from 'enzyme';
import React from 'react';
import CardList from './CardList';

it('ensures CardList rendered like snapshot', () => {
  const mockRobots = [{
    id: 1,
    name: 'unknown',
    email: 'unknown@unknown.com'
  }];

  expect(shallow(<CardList contacts={mockRobots}/>)).toMatchSnapshot();
});
