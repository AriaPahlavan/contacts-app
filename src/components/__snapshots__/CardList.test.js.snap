// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ensures CardList rendered like snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CardList
    contacts={
        Array [
            Object {
              "email": "unknown@unknown.com",
              "id": 1,
              "name": "unknown",
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Card
          email="unknown@unknown.com"
          id={1}
          name="unknown"
/>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "1",
        "nodeType": "function",
        "props": Object {
          "email": "unknown@unknown.com",
          "id": 1,
          "name": "unknown",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Card
            email="unknown@unknown.com"
            id={1}
            name="unknown"
/>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "function",
          "props": Object {
            "email": "unknown@unknown.com",
            "id": 1,
            "name": "unknown",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getSnapshotBeforeUpdate": true,
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
